include/boost/geometry/geometries/concepts/check.hpp

```diff
@@ -181,6 +181,16 @@ struct check<Geometry, multi_polygon_tag, false>
     : detail::concept_check::check<concepts::MultiPolygon<Geometry> >
 {};
 
+template <typename Geometry, bool IsConst>
+struct check<Geometry, dynamic_geometry_tag, IsConst>
+    // TODO traits::visit
```

Comment:
There are still many `TODO` 's in this PR.
I don't know how important they are, and when this should be done.

Reply:
There are many TODOs because there are many ways how to proceed. We also didn't talk about it on the Issues page. So I'm treating this PR as a ground for brainstorming.

The question here is: should we check all of the geometries stored in a dynamic geometry here or rather do this in a `variant_dispatch` (see above) and here ignore the check?

Btw, the `traits::visit` is wrong. It'd have to be a`concept_check` for `concepts::DynamicGeometry` checking all of the types defined in `traits::geometry_types<>::type`.

Reply:
There are the concepts for DG and GC now, though only the first geometry from the `geometry_types<>` sequence is checked there because we also need it to check whether or not it can be assigned to DG or emplaced_back to GC.

So I see two possibilities, as I mentioned:
- check the first type in the concept check for DG and GC and the rest when they are actually used, i.e. in the `variant_dispatch` portion of an algorithm
- check all of the types in the concept check

Do you have any preference?

Reply:
As mentioned above, I'm fine either way. If I would have a preference, I would select the option which gives the most clear error message if users pass a wrong (not fulfilling) geometry. Unless that requires a lot of extra code.

Reply:
It wouldn't require a lot of extra code. I'm not sure about the error message, It'd probably be the same but I think I'll do this in concepts to keep all of the checks in one place.

Reply:
OK

Reply:
Done.